clicommand/agent_start.go

```diff
@@ -758,6 +759,18 @@ var AgentStartCommand = cli.Command{
 		))
 		defer done()
 
+		// Verify that the bootstrap buildkite-agent executable matches the current host agent
+		// executable. In development builds, it exits on mismatch; otherwise it only logs a warning.
+		// This is to avoid confusion when making changes to the buildkite-agent but forgetting to
+		// update the buildkite-agent binary available from $PATH
+		if err := checkBinaryPaths(); err != nil {
+			if version.IsDevelopmentBuild() {
+				return fmt.Errorf("check binary paths: %s", err)
+			} else {
+				l.Warn("check binary paths: %s", err)
```

Comment:
This could create a lot of logs for customers. I wonder if it should be debug level?

Reply:
it should only be once at the start of the agent logs (not the job logs), so i think that this is probably fine.